src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -169,6 +170,59 @@ public void InitializeService(ServiceHost serviceHost)
         }
 
         /// <summary>
+        /// Handles a request to execute a string and return the result
+        /// </summary>
+        internal Task HandleExecuteAndReturnResultRequest(ExecuteAndReturnResultParams executeParams,
+            RequestContext<ExecuteAndReturnResultResult> requestContext)
+        {
+            Func<string, Task> queryCreateFailureAction = message => requestContext.SendError(message);
+            ExecuteStringParams newParams = new ExecuteStringParams
+            {
+                Query = executeParams.QueryString,
+                OwnerUri = executeParams.OwnerUri
+            };
+
+            ResultOnlyContext<ExecuteAndReturnResultResult> newContext = new ResultOnlyContext<ExecuteAndReturnResultResult>(requestContext);
+
+            // handle sending event back when the query completes
+            Query.QueryAsyncEventHandler queryComplete = async q =>
+            {
+                int rowCount = 0;
+                try
+                {
+                    rowCount = Convert.ToInt32(q.Batches[0].ResultSets[0].RowCount);
+                }
+                catch(OverflowException e)
+                {
+                    await requestContext.SendError(e);
+                }
+                SubsetParams subsetParams = new SubsetParams
+                {
+                    BatchIndex = 0,
+                    ResultSetIndex = 0,
+                    RowsStartIndex = 0,
+                    RowsCount = rowCount
+                };
+                ResultSetSubset subset = await InterServiceResultSubset(subsetParams);
+                ExecuteAndReturnResultResult result = new ExecuteAndReturnResultResult
+                {
+                    RowCount = q.Batches[0].ResultSets[0].RowCount,
+                    ColumnInfo = q.Batches[0].ResultSets[0].Columns,
+                    Rows = subset.Rows
+                };
+                await requestContext.SendResult(result);
+            };
+
+            // handle sending error back when query fails
+            Query.QueryAsyncErrorEventHandler queryFail = async (q, e) =>
+            {
+                await requestContext.SendError(e);
+            };
+
+            return InterServiceExecuteQuery(newParams, requestContext, null, queryCreateFailureAction, queryComplete, queryFail);
```

Comment:
- Should we be doing cleanup after this is done? It seems like in this case we're assuming client-side caching, so shouldn't we dispose the resultset etc? 

- I am guessing you may run multiple different queries on the same channel. If this is the case, how do we ensure they don't conflict? I'm not sure how the query service manages this - is there a queue?

Also what if the size is larger than some set limit, should we handle that at all?